File: snippets/python/ordinal.py
# snippet:
# title: "Ordinal suffix for a number"
# card_title: "Ordinal suffix"
# summary: "Return the English ordinal form of an integer, including the special cases 11th, 12th, and 13th as well as 1st, 2nd, and 3rd."
# tags: [text]
# added: "2026-08-18T19:55:36+01:00"
# submitted_by: Lupraxus
# runnable: false
# end-snippet
from typing import Dict


def ordinal(num: int) -> str:
    suffixes: Dict[int, str] = {1: 'st', 2: 'nd', 3: 'rd'}

    if 10 <= num % 100 <= 20:
        suffix = 'th'
    else:
        suffix = suffixes.get(num % 10, 'th')

    return f"{num}{suffix}"

File: snippets/rust/ordinal.rs
// snippet:
// title: "English ordinal form of a number"
// card_title: "Ordinal suffix"
// summary: "Return an integer with its English ordinal suffix, treating 11, 12, 13 and other teens as th, and 1, 2, 3 as st, nd, rd."
// tags: [text]
// added: "2026-08-21T13:27:00+01:00"
// submitted_by: Lupraxus
// runnable: false
// end-snippet
fn get_ordinal(n: u32) -> String {
    let suffix = match (n % 10, n % 100) {
        (1, 11) | (2, 12) | (3, 13) => "th",
        (1, _) => "st",
        (2, _) => "nd",
        (3, _) => "rd",
        _ => "th",
    };
    format!("{n}{suffix}")
}
